merge 2020.06

kkrebs opened the pull request.

yagee-de committed:
MCR-2276 do not include IFS1 commands by default

yagee-de committed:
MCR-2210 disable nested document handling by default

yagee-de committed:
MCR-2211 disable dynamic fields by default

bt1cu committed:
MCR-2273 Add the URI resolver to check permission (#1057)

* MCR-2273 Add the URI resolver to check permission
* MCR-2273 add support for permission without id
* MCR-2273 remove System.out.println
* MCR-2273 add XSL3 function check-permission
Co-authored-by: Thomas Scheffler <github@yagee.de>

yagee-de committed:
MCR-2279 update dependencies

rsteph-de committed:
MCR-1914 default for oai.search.restriction property (#1071)

MyCoRe-Travis commented:
Something gone wrong on fc8f4c5846576635c015c169b94bce478d27d2fe! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6034-pull_request-1068

sebhofmann committed:
MCR-2058 update javascript dependencies (#1065)

* mets-editor javascript dependencies
* viewer javascript dependencies
* classeditor dependencies
* grunt
* wcms dependencies
* webtools dependencies
* webcli javascript dependencies
* travis environment

yagee-de committed:
MCR-2275 make ifs2 the default scheme (#1062)

MyCoRe-Travis commented:
Something gone wrong on 7fc24a10af9532701069ce8b5d63ca47bbdaa684! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6041-pull_request-1068

sebhofmann committed:
MCR-2284 Fix ACL Checks

sebhofmann committed:
MCR-1583 clean up old properties from mycore urn (#1067)

Removed isAllowedObjectForURNAssignment because its application specific
code and unecessary too.Can be solved with XSL, just check if
MCR.URN.Enabled.Objects exists and contains type of the object. Get the
type with substring-before and -after.

Renamed all MCR.URN properties to MCR.PI

This makes sure that no MCR.URN.* property is used anymore

sebhofmann committed:
MCR-2281 add functions for i18n (#1066)

Add functions translate and translate-with-params

sebhofmann committed:
MCR-2058 fix_pdfjs_dependency

yagee-de committed:
set Firefox to 'latest-esr' again

kkrebs committed:
MCR-2071 change default connection configuration (#1074)

add hikari and c3p0 configuration example for production

rsteph-de committed:
MCR-2221 remove deprecated derivate attributes (#1070)

* move code for label from MCRBase to MCRObject
* replace or remove mcrDerivate.getLabel() / setLabel()

you may use a title instead of the label
mcrDerivate.getDerivate().getTitles()
    .add(new MCRMetaLangText("title", null, null, 0, null, label));

* remove command synchronize derivate with ID {0}
* remove derivate attribute display="true|false"
* set "default" title where a "default" label was set before
* do not set "data object from ..." as default title anymore
* move display access check for derivate in static method
